crates/bevy_input/src/keyboard.rs

```diff
@@ -253,3 +253,213 @@ pub enum KeyCode {
     Paste,
     Cut,
 }
+
+impl KeyCode {
+    /// Converts a key into its textual char representation on a QWERTY keyboard depending on the provided shift status.
+    ///
+    /// Calls [`KeyCode::to_qwerty_char_with_shift`] if shift is down, or [`KeyCode::to_qwerty_char_without_shift`] if shift isn't down.
+    ///
+    /// Returns `None` if the key has no valid textual mapping on a QWERTY keyboard with the provided shift status.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bevy_input::prelude::KeyCode;
+    /// #
+    /// let c1 = KeyCode::Slash.to_qwerty_char(false);
+    /// assert_eq!(c1, Some('/'));
+    ///
+    /// let c2 = KeyCode::Slash.to_qwerty_char(true);
+    /// assert_eq!(c2, Some('?'));
+    ///
+    /// let c3 = KeyCode::End.to_qwerty_char(true);
+    /// assert_eq!(c3, None);
+    /// ```
+    pub fn to_qwerty_char(self, shift_down: bool) -> Option<char> {
```

Comment:
I think we should probably remove this method. We don't need two ways to get characters with/without shift.

How about just `to_char()` and `to_char_with_shift()`